yew/src/virtual_dom/mod.rs

```diff
@@ -42,23 +42,232 @@ pub use self::vtag::VTag;
+    onload
+    onloadeddata
+    onloadedmetadata
+    onloadstart
+    onmousedown
+    onmouseenter
+    onmouseleave
+    onmousemove
+    onmouseout
+    onmouseover
+    onmouseup
+    onpause
+    onplay
+    onplaying
+    onprogress
+    onratechange
+    onreset
+    onresize
+    onscroll
+    onsecuritypolicyviolation
+    onseeked
+    onseeking
+    onselect
+    onslotchange
+    onstalled
+    onsubmit
+    onsuspend
+    ontimeupdate
+    ontoggle
+    onvolumechange
+    onwaiting
+    onwheel
+    oncopy
+    oncut
+    onpaste
+    onanimationcancel
+    onanimationend
+    onanimationiteration
+    onanimationstart
+    ongotpointercapture
+    onloadend
+    onlostpointercapture
+    onpointercancel
+    onpointerdown
+    onpointerenter
+    onpointerleave
+    onpointerlockchange
+    onpointerlockerror
+    onpointermove
+    onpointerout
+    onpointerover
+    onpointerup
+    onselectionchange
+    onselectstart
+    onshow
+    ontouchcancel
+    ontouchend
+    ontouchmove
+    ontouchstart
+    ontransitioncancel
+    ontransitionend
+    ontransitionrun
+    ontransitionstart
+}
+
 /// The `Listener` trait is an universal implementation of an event listener
 /// which is used to bind Rust-listener to JS-listener (DOM).
 pub trait Listener {
     /// Returns the name of the event
-    fn kind(&self) -> &'static str;
-    /// Attaches a listener to the element.
+    fn kind(&self) -> ListenerKind;
+
+    /// Attaches a listener to the Element
+    #[cfg(feature = "std_web")]
     fn attach(&self, element: &Element) -> EventListener;
+
+    /// Handles an event firing
+    #[cfg(feature = "web_sys")]
+    fn handle(&self, event: Event);
+
+    /// Defines flags toi modify the handling of the event. See yew::callback for more details.
+    #[cfg(feature = "web_sys")]
+    fn flags(&self) -> crate::callback::Flags;
 }
 
 impl fmt::Debug for dyn Listener {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "Listener {{ kind: {} }}", self.kind())
+        cfg_match! {
+            feature = "web_sys" => write!(
+                f,
+                "Listener {{ kind: {}, flags: {:?} }}",
+                self.kind().as_ref(),
+                self.flags()
+            ),
+            feature = "stdweb" => write!(
+                f,
+                "Listener {{ kind: {} }}",
+                self.kind().as_ref()
+            ),
+        }
+    }
+}
+
+/// A list of event listeners
+#[derive(Debug, Clone)]
+pub enum Listeners {
+    /// Added to global registry by ID
+    #[cfg(feature = "web_sys")]
+    Registered(u32),
+
+    /// Added to the Element. Stored so they are removed on VTag drop.
+    #[cfg(feature = "std_web")]
+    Registered(Rc<Vec<EventListener>>),
+
+    /// Not yet added to the element
+    Pending(Vec<Rc<dyn Listener>>),
+}
+
+impl Default for Listeners {
+    fn default() -> Self {
+        Self::Pending(Vec::new())
+    }
+}
+
+impl PartialEq for Listeners {
```

Comment:
This looks pretty nice, thanks for taking the time to do this! I think it will especially help with `children` comparison

Reply:
No problem.